<?xml version="1.0"?>
<launch>

	<!-- INITIALIZE INFINITAM: -->
    <node pkg="infinitam" name="InfiniTAM_node" type="InfiniTAM_node"  output="screen"></node>

	<!-- REMAP TOPIC NAMES: rosbags have a different topic then infotam-->	
	<remap from="/sr300/rgb/camera_info" to="/camera/rgb/camera_info"/>
	<remap from="/sr300/depth/camera_info" to="/camera/depth/camera_info"/>
	<remap from="/sr300/rgb/image_raw" to="/camera/rgb/image_raw"/>
	<remap from="/sr300/depth/image_raw" to="/camera/depth/image_raw"/>

	<arg name="arg1" default="false" />
	<arg name="arg2" default="false" />
	<arg name="arg3" default="false" />
	<arg name="arg4" default="false" />

	<!-- INITIALIZE ROSBAGS: -->
	<group if="$(arg arg1)">
		<node pkg="rosbag" type="play" name="player" output="screen" args="-d 5 /home/gocarlos/rosbags/scripts_2016-11-17-15-20-23.bag"/>
	</group>

	<group if="$(arg arg2)">
		<node pkg="rosbag" type="play" name="player" output="screen" args="-d 5 /home/gocarlos/rosbags/scripts_2016-11-17-15-26-18.bag"/>
	</group>

	<group if="$(arg arg3)">
		<node pkg="rosbag" type="play" name="player" output="screen" args="-d 5 /home/gocarlos/rosbags/scripts_2016-11-17-15-30-27.bag"/>
	</group>
	
	<group if="$(arg arg4)">
		<node pkg="rosbag" type="play" name="player" output="screen" args="-d 5 /home/gocarlos/rosbags/depth_points.bag"/>
	</group>
</launch>